packages/core-transaction-pool/src/memory.ts

```diff
@@ -138,7 +145,13 @@ export class Memory {
             this.byType[type].add(MemoryTransaction);
         }
 
-        if (MemoryTransaction.expiresAt(maxTransactionAge) !== null) {
+        if (type !== Enums.TransactionTypes.TimelockTransfer) {
+            const maxHeight: number = Crypto.slots.getHeight() + maxTransactionAge;
```

Comment:
I don't think this works, because we never update the slots height so it will always return 1. We have two options, either update the slots height when we receive a new block (see https://github.com/ArkEcosystem/core/blob/2.4/packages/core-blockchain/src/state-storage.ts#L99) or query the blockchain state (`app.resolve("state").getLastBlock().data.height`).